### ISSUE TITLE
bash/zsh completions 

### ISSUE THREAD

**Issue by threepointone** [cloudflare/workers-sdk#53]
We can leverage `yargs.completion()` to autogenerate bash/zsh completions for wrangler. I tried it, and it's... fine. Annoyingly, it hijacks any suggestions for filenames, and I assume we'll have to reimplement that. Leaving this open if anyone wants a crack at it.

**Comment by JacobMGEvans** [comment]
yargs.completion stuff from what I can tell it seems the filename completion not working is fundamental issue with their script they generate for zsh (I use zsh) I was able to get completion to work for the main commands and then the filesystem to autocomplete after the command completions, however, it overrides the flags 
https://github.com/yargs/yargs/issues/2113

**Comment by threepointone** [comment]
I don't think this is something we can work on now, it'll take a ton of work to get this to work well for us since yargs doesn't even list files etc. Should probably unassign and move this to backlog.

**Comment by petebacondarwin** [comment]
I agree. Moving to the backlog...

**Comment by DaniFoldi** [comment]
In case this gets picked up, could we have `fish` (https://fishshell.com) added to the list of supported shells?

**Comment by Cherry** [comment]
I don't think closing an issue because it's blocked upstream makes a lot of sense...

This is still something people want to see, and closing the issue is only going to cause confusion when someone searches for it later, or this needs to be addressed. What's the harm in leaving it open and just adding a `blocked` label or something so it can be easily filtered? cc @admah

**Comment by alyssadev** [comment]
Completions would certainly be nice to have as a feature. The mentioned upstream blocking issue is implemented by yargs/yargs#2331, just hasn't been merged since May.

**Comment by pauliesnug** [comment]
Any update on this?

**Comment by alyssadev** [comment]
Blocked by upstream as mentioned

**Comment by Lan-Hekary** [comment]
Any news on this front ?

**Comment by pauliesnug** [comment]
> Blocked by upstream as mentioned

@alyssadev We could do a `pnpm patch` or something similar, since its just a few LOC change?

**Comment by alyssadev** [comment]
> > Blocked by upstream as mentioned
> 
> @alyssadev We could do a `pnpm patch` or something similar, since its just a few LOC change?

I'm not involved in any way, I just wanted to help by indicating that there's an upstream issue that seems to be blocking.

**Comment by pauliesnug** [comment]
Oh I see sorry. I might make a PR with the patch!

**Comment by hkdobrev** [comment]
The upstream blocking issue is closed as old. Isn't it possible to have shell completions at least for bash without the zsh blocking issue?

**Comment by jcranendonk** [comment]
FYI the blocking issue has been merged as of April 16 🎉
